corehq/apps/reports/management/commands/project_stats_report.py

```diff
@@ -339,8 +400,200 @@ def _get_table_size(config):
             for datasource in static_datasources + dynamic_datasources
         ], key=lambda r: r[-1] if r[-1] != 'Table not found' else 0)
 
+        ResourceModel.set_stat(ResourceModel.DATASOURCES_INFO, rows)
+
+    def _case_transactions(self):
+        db_name = get_db_aliases_for_partitioned_query()[0]
```

Comment:
@proteusvacuum That is a great question! I saw that this command (`project_stats_report`) already uses only one partition to gather data, so I queried on the #gtd-tech-team slack channel about it, upon which Simon replied
> For this command specifically we are just producing estimate numbers and are assuming that data is 'evenly' distributed between the shards 

Back then I just went with this assumption, but maybe it's time to challenge it? 

Since the resource model will only really rely on the biggest contributors to the resource usage, the question I guess becomes how accurate we want those contributions estimated. 

I'm wondering now if we say, "assuming data is evenly distributed", how are we (or are we?) taking the number of shards into account. Querying one shard with evenly distributed data among 10 shards yields 1/10 of the total data.

@snopoke Any thoughts?

Reply:
You could run some queries to validate the assumption or you could run the stats query on all the shards.